src/test/java/bio/terra/service/snapshot/SnapshotServiceTest.java

```diff
@@ -968,4 +972,63 @@ public void testUpdateSnapshotDuosDatasetThrowsWhenDuosClientThrows() {
     // Job is not created or submitted if DUOS client cannot successfully obtain DUOS dataset
     verifyNoInteractions(jobBuilder);
   }
+
+  @Test
+  public void getSnapshotIdsAndRoles() throws ParseException {
+    // Arranging Sam-accessible snapshots (could contain snapshots registered in a different TDR)
+    UUID accessibleNonTdrSnapshotId = UUID.randomUUID();
+    IamRole role = IamRole.DISCOVERER;
+    Map<UUID, Set<IamRole>> resourcesAndRoles =
+        Map.of(snapshotId, Set.of(role), accessibleNonTdrSnapshotId, Set.of(role));
+    when(iamService.listAuthorizedResources(TEST_USER, IamResourceType.DATASNAPSHOT))
+        .thenReturn(resourcesAndRoles);
+
+    // Arranging RAS-accessible snapshots (ECM could throw)
+    HttpClientErrorException ecmException = new HttpClientErrorException(HttpStatus.I_AM_A_TEAPOT);
```

Comment:
haha - I have never come across this status before 😆

Reply:
Lol!  I really like it for tests where the status code itself is inconsequential.